app/frontend/src/components/EditLocationMap.tsx

```diff
@@ -120,52 +121,96 @@ export default function EditLocationMap({
     map.current!.off("touchmove", moveHandler);
     map.current!.getCanvas().style.cursor = "move";
 
-    centerCoords.current = e.lngLat;
-    redrawMap();
+    centerCoords.current = e.lngLat.wrap();
     syncCoords();
+    redrawMap();
   };
 
   const redrawMap = () => {
+    if (!isBlank.current) {
+      map.current!.setLayoutProperty("circle", "visibility", "visible");
```

Comment:
```suggestion
      map.current!.setLayoutProperty("circle", "visibility", "visible");
      isBlank.current = false;
```

Reply:
Or have I misunderstood this...

Reply:
`isBlank.current` is already `false` here?

Reply:
Oh yeah. Why are you setting visibility to visible if it's blank?

Reply:
If there's no `initalLocation`, then the map is blank (`isBlank`), so it doesn't display the circle.

As soon as you select a coordinate (by double clicking/searching and selecting option/etc), the map becomes not blank, so we show the cirlce layer

Reply:
Gotcha. Can this be done somewhere else, or are there lots of places the circle can become visible? Otherwise, it's a bit confusing, a side effect in the redraw function

Reply:
Yeah, i've moved it